internal/ethapi/api_bundle.go

```diff
@@ -141,3 +143,52 @@ type bundleError struct {
 func (e *bundleError) ErrorCode() int {
 	return InvalidBundleParamError
 }
+
+// Bundles returns the bundles in the given block range.
+// fromBlock is beginning of the queried range, must be the hex value of a block number,
+// toBlock   is end of the range, nil means latest block.
+func (s *PrivateTxBundleAPI) Bundles(ctx context.Context, fromBlock, toBlock *rpc.BlockNumber) ([]*types.BundlesItem, error) {
+	if fromBlock == nil {
+		return nil, newBundleError(errors.New("the fromBlock is required"))
+	}
+
+	if fromBlock.Int64() <= 0 {
+		return nil, newBundleError(errors.New("the fromBlock must be hex value number and greater than 0"))
+	}
+
+	from := fromBlock.Int64()
+
+	var to int64
+	if toBlock == nil || *toBlock == rpc.LatestBlockNumber || *toBlock == rpc.PendingBlockNumber {
+		to = s.b.CurrentHeader().Number.Int64()
+	} else {
+		to = toBlock.Int64()
+	}
+
+	if to < from || to >= from+types.MaxBundleAliveBlock {
+		return nil, newBundleError(errors.New("the toBlock must be greater than fromBlock and less than fromBlock + 100"))
+	}
+
+	bundleMetrics := s.b.Bundles(ctx, from, to)
+
+	ret := make([]*types.BundlesItem, 0)
+
+	for number, bundles := range bundleMetrics {
```

Comment:
we need number in order

Reply:
and need `BundlesItem` even it does not have any bundle

Reply:
fixed